### ISSUE TITLE
Noto Sans Arabic - ALEF disconnected when certain diacritics present

### ISSUE THREAD

**Issue by kmuncie** [googlefonts/noto-fonts#1555]
## Noto Sans Arabic - Alef disconnected when diacritics present

This may be related to the issues reported in #1038 

## Where the font came from and when
https://github.com/googlefonts/noto-fonts/blob/master/phaseIII_only/hinted/ttf/NotoSansArabic/NotoSansArabic-Regular.ttf (June 2019)

## Font Version
Version 2.000; ttfautohint (v1.8.2)

## Numbered Examples Shown Below

1. فَٱلـ

Char | Unicode | Description
-- | -- | --
ف | U+641 | ARABIC LETTER FEH (ARABIC LETTER FA)
َ | U+64E | ARABIC FATHA (ARABIC FATHAH)
ٱ | U+671 | ARABIC LETTER ALEF WASLA (ARABIC LETTER HAMZAT WASL ON ALEF)
ل | U+644 | ARABIC LETTER LAM
ـ | U+640 | ARABIC TATWEEL

2. يًّا

Char | Unicode | Description
-- | -- | --
ي | U+64A | ARABIC LETTER YEH (ARABIC LETTER YA)
ً | U+64B | ARABIC FATHATAN
ّ | U+651 | ARABIC SHADDA (ARABIC SHADDAH)
ا | U+627 | ARABIC LETTER ALEF

3. بٱلْـ

Char | Unicode | Description
-- | -- | --
ب | U+628 | ARABIC LETTER BEH (ARABIC LETTER BAA)
ٱ | U+671 | ARABIC LETTER ALEF WASLA (ARABIC LETTER HAMZAT WASL ON ALEF)
ل | U+644 | ARABIC LETTER LAM
ْ | U+652 | ARABIC SUKUN
ـ | U+640 | ARABIC TATWEEL

![arabic-issue-2](https://user-images.githubusercontent.com/582755/60994152-f584a200-a31d-11e9-97b5-1d8e41cb088e.jpg)

**Comment by dscorbett** [comment]
Examples 1 and 3 are covered by #1179.

**Comment by nizarsq** [comment]
Reported issue reproducible. Tested font: NotoSansArabic-Regular.ttf hinted version 2.0. 
Tested on unhinted version 2.5 issue reported not observed.
<img width="669" alt="Screen Shot 2020-05-19 at 11 56 33 PM" src="https://user-images.githubusercontent.com/22158882/82415124-e5a11880-9a2c-11ea-9945-9f8b0b6a0e2d.png">

**Comment by kmuncie** [comment]
@nizarsq please note that even though the reported issue of character connection is resolved in the unhinted version it also shows an issue in the same string with a poorly positioned diacritic ( U+64E ARABIC FATHA (ARABIC FATHAH)) above the ف (U+641 ARABIC LETTER FEH (ARABIC LETTER FA)). 

The issue of poorly positioned diacritics is pretty severe as outlined in https://github.com/googlefonts/noto-fonts/issues/1038.

It would be nice to have some movement on these issues which severely impact readability of any text displayed in Noto Sans Arabic.

**Comment by nizarsq** [comment]
@kmuncie I agree, I noticed the issue with the diacritic. I suggest to file another bug (if there is no bug already filed),  it will make it easier to track. Thanks

**Comment by kmuncie** [comment]
Just did some testing with [unhinted Noto Sans Arabic in phaseIII ](https://github.com/googlefonts/noto-fonts/blob/master/phaseIII_only/unhinted/ttf/NotoSansArabic/NotoSansArabic-Regular.ttf)which is sitting at v2.500 and it appears a huge redesign is taking place. It is extremely unfinished so I assume it is a work in progress and not at all ready for general usage. 

```
PostScript name:     NotoSansArabic-Regular
Version:             Version 2.500
Unique ID:           2.500;GOOG;NotoSansArabic-Regular
Description:         Designed by Mohamad Dakak
Designer:            Mohamad Dakak (based on work by Nadine Chahine - Monotype Design Team)
Designer URL:        mhmd.dakak@gmail.com
Manufacturer:        Mohamad Dakak
```

**@marekjez86 can you share anything about these changes and the roadmap for Arabic?** I do not want to submit further bug reports if the entire font is being reworked and bug reports for v2.000 won't help. Thank you.

**Comment by marekjez86** [comment]
@kmuncie : in what sense doesn't seem to be unfinished?
I'm asking because at this point, I'm out of money, for this font, so it will stay as it is right now :-) [unless I get some open source volunteers].

Feel free to file more bugs, if you see any issues.

**Comment by kmuncie** [comment]
Here is a short sample string to exhibit some of the problems, there are more but this is a start: ` فِي اَلْعَرَبِيَّةُ اَلْآيَةُ ٱلسَّنَوِيَّةُ لِعَامِ ٢٠٢٠‎`

![arabic-2_500-issues](https://user-images.githubusercontent.com/582755/83194557-aee29680-a0ed-11ea-94e4-4760c5dec05d.jpg)

- **Red arrows** = bugs which make the text practically unreadable
- **Orange arrows** = dramatic design changes between the two versions I would not have expected in a dot release upgrade

It is curious why well documented bugs in v2.000 linger while a larger redesign seems to be introducing even more regressions, at least at this stage of its development. Would the project be open to contributions to resolve bugs in v2.000 or will they not be taken because of this in-progress redesign? Thanks for your time @marekjez86

**Comment by nizarsq** [comment]
I looked at the unhinted Noto Sans Arabic and I may list the issues I noticed here:
1- Misplaced diacritics. Diacritics are sitting too low, this causes overlapping with high characters, dots or other diacritics. I believe this issue should be fixed at some point (Arabic speakers relay on diacritics in reading and writing)

2- Another issue (Not sure if it is intended) unhinted font looks a little bigger relative to the hinted font with the same weight.

3- Beside the size compared with hinted font, some characters in the unhinted fonts are not consistent. I noticed characters with circular or oval shapes (half circular, oval shape too) look bigger compared with the rest of the characters in the unhinted font itself. The following characters are examples of these characters (U+0639 (ARABIC LETTER AIN), U+0645 ARABIC LETTER MEEM and  U+0641 (ARABIC LETTER FEH ARABIC LETTER FA))

4- There are some other minor issues like missing the ligature for U+0644 (ARABIC LETTER LAM) and U+0627 (ARABIC LETTER ALEF when they are joined).

I believe filing individual bugs for those issues will help in tracking them in future.
<img width="1321" alt="Screen Shot 2020-05-29 at 12 15 38 AM" src="https://user-images.githubusercontent.com/22158882/83233941-55f21d00-a144-11ea-9331-75fac3fbd517.png">
